torch/_inductor/scheduler.py

```diff
@@ -138,6 +138,45 @@ def prune_deps(self):
             if dep.name not in self.scheduler.available_buffer_names
         }
 
+    def prune_redundant_deps(self, name_to_fused_node):
+        """
+        Prunes stardeps intended for mutation ordering
+        on an upstream fused node if after fusion there is another dependency
+        on the fused upstream node, making the stardep redundant
+
+        In essence this enforces an ordering on fusions. As fusions occur, prunable stardeps will
+        be incrementally removed, enabling other fusions, ensuring they are fused in order.
+        """
+        name_to_dep_count = collections.Counter()
+
+        for dep in self.unmet_dependencies:
+            name_to_dep_count[name_to_fused_node[dep.name].get_name()] += 1
+
+        def should_prune(dep):
+            if isinstance(dep, WeakDep):
+                is_redundant = (
+                    name_to_dep_count[name_to_fused_node[dep.name].get_name()] > 1
```

Comment:
This will be buggy if we have two WeakDep pointing to the same node.  In this case both will be removed.

Perhaps exclude WeakDep from `name_to_dep_count` and check `>0`